src/layoutparser/models/paddledetection/layoutmodel.py

```diff
@@ -0,0 +1,272 @@
+
+        This function is a general step to support multiple backends in the layout-parser
+        library.
+
+        Args:
+            path (str): The given input path that might or might not contain the detector name.
+
+        Returns:
+            str: a modified path that contains the detector name.
+        """
+        if path.startswith("lp://"):  # TODO: Move "lp://" to a constant
+            model_name = path[len("lp://") :]
+            model_name_segments = model_name.split("/")
+            if (
+                len(model_name_segments) == 3
+                and "paddleDetection" not in model_name_segments
+            ):
+                return "lp://" + self.DETECTOR_NAME + "/" + path[len("lp://") :]
+        return path
+
+    def load_predictor(self,
+                    model_dir,
+                    batch_size=1,
+                    enforce_cpu=False,
+                    enable_mkldnn=True,
+                    thread_num=10):
+        """set AnalysisConfig, generate AnalysisPredictor
+        Args:
+            model_dir (str): root path of __model__ and __params__
+            enforce_cpu (bool): whether use cpu
+        Returns:
+            predictor (PaddlePredictor): AnalysisPredictor
+        Raises:
+            ValueError: predict by TensorRT need enforce_cpu == False.
+        """
+
+        config = self._inference.Config(
+            os.path.join(model_dir, 'inference.pdmodel'),
+            os.path.join(model_dir, 'inference.pdiparams'))
+
+        if not enforce_cpu:
+            # initial GPU memory(M), device ID
+            # 2000 is an appropriate value for PaddleDetection model
+            config.enable_use_gpu(2000, 0)
+            # optimize graph and fuse op
+            config.switch_ir_optim(True)
+        else:
+            config.disable_gpu()
+            config.set_cpu_math_library_num_threads(thread_num)
+            if enable_mkldnn:
+                config.enable_mkldnn()
+
+        # disable print log when predict
+        config.disable_glog_info()
+        # enable shared memory
+        config.enable_memory_optim()
+        # disable feed, fetch OP, needed by zero_copy_run
+        config.switch_use_feed_fetch_ops(False)
+        predictor = self._inference.create_predictor(config)
+        return predictor
+
+    def preprocess(self, image):
+        """ preprocess image
+        Args:
+            image (np.ndarray): image (np.ndarray)
+        Returns:
+            inputs (dict): input of model
+        """
+        # read rgb image
+        image, im_info = decode_image(image, self.im_info)
+        # resize image by target_size and max_size
+        image, im_info = resize(image, im_info)
+        # normalize image
+        image, im_info = normalize_image(image, im_info)
+        # transpose images
+        image, im_info = permute(image, im_info)
+
+        inputs = {}
+        inputs['image'] = np.array((image, )).astype('float32')
```

Comment:
This feels weird -- probably it's better to have `np.array(image)[np.newaxis, :].astype('float32')`